packages/flutter_goldens/test/skia_client_test.dart

```diff
@@ -0,0 +1,632 @@
+  });
+
+  setUp(() {
+    workDirectory = fs.directory('/workDirectory')
+      ..createSync(recursive: true);
+    skiaClient = SkiaGoldClient(
+      workDirectory,
+      fs: fs,
+      process: process,
+      platform: platform,
+      httpClient: fakeHttpClient,
+    );
+  });
+
+  test('web HTML test', () async {
+    platform = FakePlatform(
+        environment: <String, String>{
+          'GOLDCTL': 'goldctl',
+          'FLUTTER_ROOT': _kFlutterRoot,
+          'FLUTTER_TEST_BROWSER': 'Chrome',
+          'FLUTTER_WEB_RENDERER': 'html',
+        },
+        operatingSystem: 'macos'
+    );
+    skiaClient = SkiaGoldClient(
+      workDirectory,
+      fs: fs,
+      process: process,
+      platform: platform,
+      httpClient: fakeHttpClient,
+    );
+
+    final File goldenFile = fs.file('/workDirectory/temp/golden_file_test.png')
+      ..createSync(recursive: true);
+
+    const RunInvocation goldctlInvocation = RunInvocation(
+      <String>[
+        'goldctl',
+        'imgtest', 'add',
+        '--work-dir', '/workDirectory/temp',
+        '--test-name', 'golden_file_test',
+        '--png-file', '/workDirectory/temp/golden_file_test.png',
+        '--passfail',
+        '--add-test-optional-key', 'image_matching_algorithm:fuzzy',
+        '--add-test-optional-key', 'fuzzy_max_different_pixels:20',
+        '--add-test-optional-key', 'fuzzy_pixel_delta_threshold:4',
+      ],
+      null,
+    );
+    process.processResults[goldctlInvocation] = ProcessResult(123, 0, '', '');
+
+    expect(
+      await skiaClient.imgtestAdd('golden_file_test.png', goldenFile),
+      isTrue,
+    );
+  });
+
+  test('isFlaky sets right args', () async {
+    platform = FakePlatform(
+        environment: <String, String>{
+          'GOLDCTL': 'goldctl',
+          'FLUTTER_ROOT': _kFlutterRoot,
+          'FLUTTER_TEST_BROWSER': 'Chrome',
+          'FLUTTER_WEB_RENDERER': 'html',
+        },
+        operatingSystem: 'macos'
+    );
+    skiaClient = SkiaGoldClient(
+      workDirectory,
+      fs: fs,
+      process: process,
+      platform: platform,
+      httpClient: fakeHttpClient,
+    );
+
+    final File goldenFile = fs.file('/workDirectory/temp/golden_file_test.png')
+      ..createSync(recursive: true);
+
+    const RunInvocation goldctlInvocation = RunInvocation(
+      <String>[
+        'goldctl',
+        'imgtest', 'add',
+        '--work-dir', '/workDirectory/temp',
+        '--test-name', 'golden_file_test',
+        '--png-file', '/workDirectory/temp/golden_file_test.png',
+        '--passfail',
+        '--add-test-optional-key', 'image_matching_algorithm:fuzzy',
+        '--add-test-optional-key', 'fuzzy_max_different_pixels:1000000000',
+        '--add-test-optional-key', 'fuzzy_pixel_delta_threshold:1020',
+      ],
+      null,
+    );
+    process.processResults[goldctlInvocation] = ProcessResult(123, 0, '', '');
+
+    expect(
+      await skiaClient.imgtestAdd('golden_file_test.png', goldenFile, isFlaky: true),
```

Comment:
Is there a way to check that "markedFlaky" was set?

Reply:
Also added, thanks!